### PR TITLE
Updating instructor notes and lesson front page re docker changes (Fixes #87)

### LINKED ISSUE #87: Checking Lesson before teaching for Docker changes
From the March 2021 Maintainer meeting we thought it would be good to add a note on in instructor notes suggests that instructors should check ahead of running the lesson and add box to front of the lesson warning that Docker the other software changes and to file an issue if something doesn't work as expected. Reminder that it needs to be tested on macOS, Windows and Linux as behaviour may differ.  @jcohen02 volunteered to make these changes

### PR DESCRIPTION
**Pull request by jcohen02**
This PR addresses issue #87, adding some text to the instructor notes in a new section "_Before teaching this lesson_" and a callout to the index page highlighting that differences can occur between the information shown in the lesson and the actual output/interface provided by Docker. Both new blocks of text invite either instructors or participants to highlight any problems they notice through opening a new issue.

### CODE REVIEW

**Comment by dme26** [comment]
This looks good to me, but another review would be good for completeness.

File: _extras/guide.md [hunk 1]
```diff
@@ -2,6 +2,29 @@
...
+
+If you experience any issues, please [open an issue] in the lesson repository
+describing the problem and platform(s) affected. The lesson maintainers will
+aim to resolve the issue as soon as possible but we also welcome the opening
+of pull requests (linked to issues) that resolve anything that doesn't work as
+expected with the lesson content.
+
+[Docker]: https://www.docker.com/
+[Docker release notes]: https://docs.docker.com/release-notes/
+[open an issue]: https://github.com/carpentries-incubator/docker-introduction/issues/new
```
Comment (sstevens2) [inline _extras/guide.md:26]: Suggest moving these to the bottom for consistency with the rest of the repo

File: index.md [hunk 2]
```diff
@@ -16,6 +16,24 @@ This session aims to introduce the use of Docker containers with the goal of usi
...
+>
+> While we do our best to ensure that this lesson remains up to date and the
+> descriptions and outputs shown match what you will see on your own computer,
+> inconsistencies can occur.
+> 
+> If you spot inconsistencies or encounter any problems, please do report them
+> by [opening an issue] in the [GitHub repository] for this lesson.
+> 
+> [opening an issue]: https://github.com/carpentries-incubator/docker-introduction/issues/new
+> [GitHub repository]: https://github.com/carpentries-incubator/docker-introduction
```
Comment (sstevens2) [inline index.md:33]: Since these links are reused, maybe instead of putting them at the bottom they should be added to `links.md`?

**Review by sstevens2 — APPROVED** [review]
Looks great!   I'd recommend moving the links to the bottom of the page referenced or `links.md` since they are mostly reused in both locations.

**Pull request merged**